src/lib/utils.jl

```diff
@@ -99,6 +99,26 @@ macro showgrad(x)
     end)
 end
 
+"""
+jacobian(f,x)
+
+Construct the Jacobian of `f` where `x` is a real-valued array
+and `f(x)` is also a real-valued array.
+"""
+function jacobian(f,x)
+    y,back  = Zygote.pullback(f,x)
+    k  = length(y)
+    n  = length(x)
+    J  = Matrix{eltype(y)}(undef,k,n)
+    e_i = fill!(similar(x), 0)
```

Comment:
`zero(x)` would be clearer

Reply:
But not mutable -- see earlier suggestion.

Reply:
```suggestion
    e_i = fill!(similar(y), 0)
```
For reverse diff, your `e_i` is the wrong size unless your jacobian happens to be square.